scripts/runTypewriter.ps1

```diff
@@ -0,0 +1,150 @@
+    [parameter(Mandatory = $false)][string]$language,
+    [parameter(Mandatory = $false)][string]$verbosity,
+    [parameter(Mandatory = $false)][string]$docs,
+    [parameter(Mandatory = $false)][string]$outputMetadataFileName,
+    [parameter(Mandatory = $false)][string]$endpointVersion,
+    [parameter(Mandatory = $false)][string]$properties,
+    [parameter(Mandatory = $false)][string]$transform,
+    [parameter(Mandatory = $false)][string]$output,
+    [parameter(Mandatory = $false)][string]$generationMode,
+    [parameter(Mandatory = $false)][bool]$cleanup = $true
+)
+
+# VARIABLES
+$gh_owner_and_repo = 'microsoftgraph/MSGraph-SDK-Code-Generator'    # Org and repo where typewriter.exe is released.
+$tempDir           = '.\temp'
+$LASTEXITCODE      = $null
+$typewriterZipDrop = "$tempDir\typewriter.zip"
+$typewriterFilesDir = "$tempDir\Release\*"
+$typewriter = Join-Path $tempDir "typewriter.exe"
+
+# Create our temporary working directory.
+if ((Test-Path -PathType Container -Path $tempDir)) {
+    # Remove-Item $tempDir -Force -Recurse
+    # Write-Host "Deleted $tempDir"
+    Write-Host "$tempDir already exists"
+} else {
+    New-Item -Path $tempDir -Type Directory
+    Write-Host "Created $tempDir"
+}
+
+# Create the typewriter.exe command line arguments from script arguments.
+$options = @()
+
+if ($verbosity)              { $options += "-v"; $options += "$verbosity" }
+if ($metadata)               { $options += "-m"; $options += "$metadata"}
+if ($language)               { $options += "-l"; $options += "$language"}
+if ($docs)                   { $options += "-d"; $options += "$docs"}
+if ($outputMetadataFileName) { $options += "-f"; $options += "$outputMetadataFileName"}
+if ($endpointVersion)        { $options += "-e"; $options += "$endpointVersion"}
+if ($properties)             { $options += "-p"; $options += "$properties"}
+if ($transform)              { $options += "-t"; $options += "$transform"}
+if ($output)                 { $options += "-o"; $options += "$output"}
+if ($generationMode)         { $options += "-g"; $options += "$generationMode"}
+
+if ((Test-Path -Path $typewriter -PathType leaf) -ne $true)
+{
+    # Get information about the GitHub releases.
+    $feedQuery = "https://api.github.com/repos/$gh_owner_and_repo/releases"
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    $jsonObject = Invoke-WebRequest -Uri $feedQuery -UseBasicParsing | ConvertFrom-Json
+
+    # Download typewriter from the latest GitHub release.
+    if ($jsonObject.assets[0].name -eq 'typewriter.zip') {
+        $downloadURL = $jsonObject.assets[0].browser_download_url # GitHub release API provides the latest
+        Invoke-WebRequest -Uri $downloadURL -OutFile $typewriterZipDrop -UseBasicParsing -Verbose | Write-Host
+    }
+    else {
+        Write-Error 'typewriter.zip was not found using the release API. Check the release on GitHub. Make sure that
```

Comment:
shouldn't we terminate if this happens instead of continuing on to expand the archive.  I can't recall if Write-Error actually terminates so I could just be missing that.

Reply:
Yes! Thank you